librad/src/identities/generic.rs

```diff
@@ -0,0 +1,543 @@
+// This file is part of radicle-link
+// <https://github.com/radicle-dev/radicle-link>
+//
+// Copyright (C) 2019-2020 The Radicle Team <dev@radicle.xyz>
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 or
+// later as published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+#![allow(clippy::type_complexity)]
+
+use std::{
+    fmt::{Debug, Display},
+    marker::PhantomData,
+    ops::Deref,
+};
+
+use serde::ser::SerializeStruct;
+
+use super::{delegation::Delegations, sealed, sign::Signatures, urn::Urn};
+
+pub mod error;
+
+#[cfg(test)]
+pub(crate) mod gen;
+#[cfg(test)]
+pub(crate) mod tests;
+
+/// The identity document, carrying metadata `T` and trust delegations `D`.
+///
+/// In `git`, this is represented as a `blob`, where the previous revision
+/// `replaces` is a `tree` oid.
+#[derive(Clone, Debug, PartialEq, serde::Deserialize)]
+pub struct Doc<T, D, Revision> {
+    /// Protocol version. Always serialised as `0` (zero).
```

Comment:
Das Spezifikation says that the protocol version is zero. I wanted to avoid `new` and builders and [s|g]etters just because of that. We could also omit the field entirely, but then we'll have to remember to add it if and when the version is zero+1. So, whatever you put there, it'll come out as zero :)